File: src/film/documentary/documentary.cpp
#include "documentary.h"
#include "../../utils/io/io.h"
#include "../../utils/random/random.h"
#include <cstdio>

Documentary::Documentary(const char *title, int year, int duration) : Film(title, year), duration(duration) {}

Film *Documentary::In(const char *title, int year, FILE *input) {
  int duration = ReadIntIn(input, 45, 3600,"Incorrect duration: expected an integer in [45, 3600]");
  return new Documentary(title, year, duration);
}

Film *Documentary::InStochastic() {
  return new Documentary(
      RandomString(RandomInt(1, 51)),
      RandomInt(1970, 3022),
      RandomInt(45, 3600));
}

void Documentary::Out(FILE *output) {
  fprintf(output, "Documentary{title=%s,year=%d,duration=%d)", title, year, duration);
}

File: src/film/documentary/documentary.py
import random

from src.film.film import Film
from src.utils import io, rand


class Documentary(Film):
    min_duration = 10
    max_duration = 3600

    def __init__(self):
        super().__init__()
        self.duration = Documentary.min_duration

    def read_from_file(self, file):
        super(Documentary, self).read_from_file(file)
        self.duration = io.read_int(file, "Documentary duration",
                                       Documentary.min_duration,
                                       Documentary.max_duration)

    def fill_stochastic(self):
        super(Documentary, self).fill_stochastic()
        self.duration = random.randint(Documentary.min_duration, Documentary.max_duration)

    def write_to_file(self, file):
        file.write('Documentary{' + F'title={self.title},year={self.year},duration={self.duration}' + '}')
